lib/registry.py

```diff
@@ -557,8 +563,9 @@ def request(self, method, url, statuses={200}, out=None, **kwargs):
          out.start(length)
          for chunk in res.iter_content(ch.HTTP_CHUNK_SIZE):
             out.write(chunk)
+            m.update(chunk)
       # Done.
-      return res
+      return res, m
```

Comment:
This interface change would change the basic contract of the method. Right now it cannot fail: either it returns a valid response, or it exits the program with an error. This change adds the possibility the method can fail, so every caller would have to deal with that. I suggest instead adding a keyword argument that is the expected digest, and `request()` does the checking.

There is also a subtle interaction with `out` to consider.